### DISCUSSION TITLE
Raspberry Pi Pico 2

### DISCUSSION THREAD

**Discussion by NuclearPhoenixx** [OpenGammaProject/Open-Gamma-Detector#60] [General]
The Raspberry Pi Pico 2 is a huge improvement to the project. Most importantly the processor will be a lot faster (new architecture and FPU), double the RAM and they finally fixed the ADC! 🚀

Read the full post on my homepage: https://nuclearphoenix.xyz/2024/08/10/pico-2

Let me know what you think, either here directly under this post or on my homepage! 👍🏻

**Comment by dc1rdb** [comment]
That's great news! Looking forward to your test results, especially w.r.t ADC performance. It wiil be a p.i.t.a. to exchange the Pico on the OGD board, though.

**Reply by NuclearPhoenixx** [reply]
Yes, you'd need a hot air station to do that. No chance you could do it with just a soldering iron.

**Comment by NuclearPhoenixx** [comment]
Just tested the board and the results are pretty amazing actually.

The DNL issues, are obviously gone and with it the `cps correction` setting and some of the code in the event interrupt that had to handle the 4 error channels. Finally no more random peaks or missing channels in the spectrum! 🎉

Due to this combined with the more powerful processor cores (and FPU), the total dead time reduced significantly, which is probably the biggest change. With the same settings both on the Pico 1 and Pico 2 at default settings the dead time in energy mode halved (!) from ~24µs to ~12µs. The dead time in Geiger mode actually reduced more than that, it only registered ~1µs of dead time on the Pico 2, while the Pico 1 had about 4µs. This is of course only the software-measured dead time, there will still be a bit of a constant overhead on top of all of that, that cannot be measured in software. It's pretty safe to assume that this hasn't changed too much between the revisions, so the relative decrease in dead time is very significant.

The flash size is now triple that of the old version at 3 MB reserved for custom spectrum storage. This means you're able to record almost 3x the amount of spectra before running out of space, which is also huge.

The cherry on top of all of this is that power efficiency went up too. Without any further changes, the detector now consumes a good 10% less power than before according to my testing. I will also investigate the new sleep/dormant modes of the RP2350 as soon as there is support for these.

On a side note, I was able to overclock the Pico 2 to 250 MHz at the default 1V core voltage in the sketch, which is 25 MHz higher than with the Pico 2 (225 MHz). Therefore the `overclock` uf2 file will be clocked at 250 MHz from now on.

All of the hardware and software files are up to date now for the Pico 2 and development will continue on the Pico 2 from now on. The new board revision 4.2 is unchanged from the old one besides the switch to the Pico 2 and an updated BOM position for the out-of-stock buzzer. I will update the post on my website to also reflect these results (link above).

**Comment by dc1rdb** [comment]
Ok, here is how I removed the old Pico from the OGD board:
Using a Dremel with a cutting disc, I carefully cut the old Pico board along the pin holes. Then I removed the center portion of the old Pico board. With only the two remaining stripes of the Pico still attached to the main board, it was no problem lifting the stripes with tweezers and desoldering them pad by pad.
Fortunately, nothing got damaged on the main board.
After soldering the Pico 2 and loading it with the 4.3.0 firmware, everything worked fine. I chose to not overclock, so my Pico 2 is running at 150 MHz. Dead times are 1 us in Geiger mode and 11 us in spectrum mode. At first glance, the spectrum looks cleaner than with the old Pico.
More detailed tests regarding FWHM and max count rate will follow.
